### PR TITLE
Added the new martial art, Slipping Carp, for traitor janitors

### PR DESCRIPTION
**Pull request by Cruix**
I coded this ages ago but dropped it because it was a little ridiculous. But I may as well jump on the traitor item pile that has appeared recently.

Slipping carp makes you immune to slips and unable to use guns at all times. While holding a mop in your active hand, you can deflect projectiles, wet floors and slip people on them instantly, instantly aggressively grab people, and if you disarm someone, the item is thrown three tiles away. Your attacks with the mop also deal more damage and ignore armor.

:cl:
rscadd: Traitor janitors can now purchase the Slipping Carp scroll to become masters of mop-based combat.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
i didn't know you could meme I think you're an amazing maintainer now.


wait who merges your PRs there's noone to do it.

**Comment by ghost** [comment]
he does

**Comment by ghost** [comment]
maintainers cant merge their own PR you fucking knob

**Comment by Super3222** [comment]
> you **fucking** knob

**Comment by Cruix** [comment]
If I never merged my own PRs, there would still be some left from when I started.

**Comment by ghost** [comment]
get more head coders

**Comment by Cruix** [comment]
We lose them faster than we can find them.

**Comment by ghost** [comment]
you have time and even me if you want memes

File: code/datums/martial.dm [hunk 1]
```diff
@@ -8,13 +8,18 @@
...
+ 	return FALSE
+
+/datum/martial_art/proc/try_deflect_projectile(mob/living/carbon/human/user, obj/item/projectile/Proj, silent = FALSE)
 	if(user.incapacitated() || (user.dna && user.dna.check_mutation(HULK)))
 		return FALSE
 	if(prob(deflection_chance))
-		user.visible_message("<span class='danger'>[user] deflecs the projectile; they can't be hit with ranged weapons!</span>", "<span class='userdanger'>You deflect the projectile!</span>")
-		playsound(user, pick("sound/weapons/bulletflyby.ogg","sound/weapons/bulletflyby2.ogg","sound/weapons/bulletflyby3.ogg"), 75, 1)
+		if(!silent)
+			user.visible_message("<span class='danger'>[user] deflecs the projectile; they can't be hit with ranged weapons!</span>", "<span class='userdanger'>You deflect the projectile!</span>")
```
Comment (ShadowDeath6) [inline code/datums/martial.dm]: deflec

Comment (ghost) [reply code/datums/martial.dm]: but he also attac

Comment (Cruix) [reply code/datums/martial.dm]: I anticipated this meme almost a year ago and I did not even know it.

**Comment by Amelia0010** [comment]
> We lose them faster than we can find them.

@AshCorr for headcoder when

**Comment by Chuggachar** [comment]
Seems OP, you can just ignore any ranged attacks and then immediately slip anyone who gets within melee range?

**Comment by Oakboscage** [comment]
Just fix regular janitor and traitor janitors get their gimmick by default.

**Comment by Cruix** [comment]
It is just a weaker version of sleeping carp, which can do almost the same thing.

**Comment by Chuggachar** [comment]
Sleeping carp isn't avalible to traitors here though is it?

**Comment by Cruix** [comment]
It is available to the sleeping carp gang, which is enabled here.

**Comment by Kmc2000** [comment]
Yes. 

Merge immediately

**Comment by ghost** [comment]
sleeping carp used to be a traitor item altogether chug

File: code/datums/martial.dm (hunk 1 above)
Comment (Super3222) [reply code/datums/martial.dm]: Please leave that in like that

**Comment by Super3222** [comment]
I'm the guy that removed sleeping carp from normal uplink. If this becomes "too OP" than I suggest making it so the janitors need the mop in their active hand to deflect projectiles, and when a projectile is deflected it swaps their active hand. Otherwise it's not that big of a deal getting into close range with a traitor that's using this item, because the mop basically becomes a stunbaton and in that regard it's just a stunbaton v stunbaton duel. My second suggestion is to make it so janitors cannot deflect anything that attacks them from behind. You can track this using `check_for_positions` which is currently attached to weapons only (in weapons.dm), but you can delete the path to make it a global proc (no big deal since tons of unsorted procs are like that).

They also can't use guns so that nullifies the use of disarming a taser with any other reason of just getting it out of the officers hands. The only problem I can see with this is what if the jaintor is carrying a riot shield which I say look at my second suggestion.

**Comment by ghost** [comment]
Tbh, a couple pissed off engineers with magboots could defeat slipping carp easily

**Comment by Kmc2000** [comment]
Why is this not merged yet

**Comment by Cruix** [comment]
Waiting for all the other traitor items to be fixed so we can have a silly traitor weapon testmerge

**Comment by KrabSpider** [comment]
PLEASE MERGE ASH!

**Comment by Moja10** [comment]
MERGE THIS

**Comment by Jc1469** [comment]
:thinking:

**Comment by Amelia0010** [comment]
🤔

**Pull request merged**